app/models/financialDetails/DocumentDetail.scala

```diff
@@ -221,9 +221,14 @@ case class DocumentDetail(taxYear: Int,
 
 case class DocumentDetailWithDueDate(documentDetail: DocumentDetail, dueDate: Option[LocalDate],
                                      isLatePaymentInterest: Boolean = false, dunningLock: Boolean = false,
-                                     codingOutEnabled: Boolean = false, isMFADebit: Boolean = false)(implicit val dateService: DateServiceInterface) {
+                                     codingOutEnabled: Boolean = false, isMFADebit: Boolean = false,
```

Comment:
we should really avoid use of boolean flags, as I understand only one of these will be "true", so these need to be modelled in the different way